packages/components/FocusZone/macos/RCTFocusZone.m

```diff
@@ -45,6 +45,21 @@ static inline CGFloat GetMinDistanceBetweenRectVerticesAndPoint(NSRect rect, NSP
 	);
 }
 
+static NSView *GetFirstKeyViewWithin(NSView *parentView)
+{
+	for (NSView *view in [parentView subviews]) {
+		if ([view canBecomeKeyView]) {
+			return view;
+		}
+
+		NSView *match = GetFirstKeyViewWithin(view);
+		if (match) {
+			return match;
+		}
```

Comment:
Re: @HeyImChris 's comment last time (https://github.com/microsoft/fluentui-react-native/pull/1400#discussion_r799043491):

Verify that this depth first search algorithm is accurate to what Apple does internally with a simple test app.

Reply:
I tested this with a simple test inside of RN-Tester. The yellow boxes are focusable. If it was depth-first, then the first yellow box would get focused. If it was breadth-first, then the second yellow box would get focused. In both cases (vertical and horizontal), the first yellow box was focused, implying depth first.

```
class KeyViewExample extends React.Component<{}> {
  render() {
    return (
      <View style={{padding: 10}}>
        <View style={styles.outer}>
          <View style={styles.middle}>
            <View style={styles.innerFocusable} focusable={true} />
          </View>
        </View>
        <View style={styles.outer}>
          <View style={styles.innerFocusable} focusable={true} />
        </View>
      </View>
    );
  }
}
```

![Screen Shot 2022-02-04 at 11 32 49 AM](https://user-images.githubusercontent.com/6722175/152575848-f5fa375b-0751-457c-82bb-816e1f217c92.png)
![Screen Shot 2022-02-04 at 11 33 08 AM](https://user-images.githubusercontent.com/6722175/152575854-4598acb6-f92f-4694-8a0d-bf3e48bf405e.png)